app/main.py

```diff
@@ -1,3 +1,16 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    cached_calcs = {}
+
+    def wrapper(*args, **kwargs) -> int | float:
```

Comment:
You can't be sure about typings here, so I'd suggest make it generic or omit them